### ISSUE TITLE
NullPointerException when calling methods on spy of a final class with JDK > 11

### ISSUE THREAD

**Issue by JanMosigItemis** [mockito/mockito#2202]
Hi there,

I may have encountered a bug in mockito-inline with respect to spies of final classes. You may want to checkout this demo project: https://github.com/JanMosigItemis/mockito.inline.spy.bug
  
I have:  
* Windows 10.0.19042.804
* Git Bash 2.30.0.windows.2 
* Maven 3.6.3
* Surefire 3.0.0-M5
* Mockito-Inline 3.7.7
* JUnit 5.7.0
* -Dnet.bytebuddy.experimental=true
* mock-maker-inline
  
The following test runs into an NPE:  
```java
import org.junit.jupiter.api.Test;
import org.mockito.Mockito;

import java.lang.reflect.Method;

/**
 * Unit test for simple App.
 */
public class AppTest {
    @Test
    public void test_something() throws Exception {
        Method realMethod = getClass().getMethod("some_method");
        Method spyMethod = Mockito.spy(realMethod);
        spyMethod.invoke(this); // <- NPE
    }

    public void some_method() {

    }
}
```
  
I cannot pin point the exact cause of the NPE from the logs / stack traces. The NPE only happens when running a maven build with JDK >11. I tested with JDK12 and 15 (same result for both). However, the test runs successfully with JDK 11. Using a mock instead of a spy works on all JDKs.  
  
Also running the test from within Eclipse (4.18.0) works, even if Eclipse is running with JDK12.

Any thoughts on this?

**Comment by TimvdLippe** [comment]
`java.lang.reflect.Method` is one of the building blocks of Mockito and therefore we can't guarantee correctness when mocking/spying such classes. Instead, we recommend using real objects. For more information, please see the full discussion in https://github.com/mockito/mockito/issues/2026#issuecomment-741663017 This will likely be addressed in #1833 to notify users when they run into these kind of situations.

**Comment by JanMosigItemis** [comment]
Thank you very much. I did not know that creating a spy will change the behavior of all instances of that type.
